napari/_qt/_qapp_model/injection/_qprocessors.py

```diff
@@ -99,7 +100,11 @@ def _add_layer_data_to_viewer(
         if layer_name:
             with suppress(KeyError):
                 # layerlist also allow lookup by name
-                viewer.layers[layer_name].data = data
+                layer = viewer.layers[layer_name]
+                layer.data = data
+                if isinstance(layer, Image):
+                    layer.reset_contrast_limits_range()
+                    layer.reset_contrast_limits()
```

Comment:
Maybe this will be a good option to not reset range when someone adjusts contrast limits manually:
```suggestion
                    if layer.contrast_limits == layer.contrast_limits_range:
                        layer.reset_contrast_limits_range()
                        layer.reset_contrast_limits()
```

Reply:
@jni @DragaDoncila @psobolewskiPhD ?